index.js

```diff
@@ -158,51 +185,59 @@ Buffer.from
 
     // if another process in currently creating the channel
     } else if (result.value === STATE_CREATING_CHANNEL) {
-      // TODO: poll for channelId
+      // poll for channelId
+      while (self.outgoingChannel.getMax().value !== 2) { await sleep(5000) }
+      channelId = self.outgoingChannel.getMax().data
     }
 
     // TODO: recreate channel if it doesn't exist
     self.outgoingPaymentChannelId = channelId
-    self.outgoingPaymentChannel = await self.api.ledger.getPaymentChannel(channelId)
+    self.outgoingPaymentChannel = await self.api.getPaymentChannel(channelId)
 
     // query peer until they have a channel id
     while (true) {
-      self.incomingPaymentChannelId = ctx.rpc.call('ripple_channel_id', self.prefix, [])
+      self.incomingPaymentChannelId = await ctx.rpc.call('ripple_channel_id', self.prefix, [])
       if (self.incomingPaymentChannelId) break
       await sleep(5000)
     }
 
     // look up channel on ledger
     // TODO: if incoming channel doesn't exist, throw an error
-    self.incomingPaymentChannel = await self.api.ledger.getPaymentChannel(self.incomingPaymentChannelId)
+    self.incomingPaymentChannel = await self.api.getPaymentChannel(self.incomingPaymentChannelId)
```

Comment:
will this throw an error if the channel doesn't exist?

Reply:
yeah, it does. I added a log message.